src/middlewared/middlewared/plugins/pool_/dataset.py

```diff
@@ -881,6 +881,9 @@ async def do_delete(self, id_, options):
         `recursive` will also delete/destroy all children datasets.
         `force` will force delete busy datasets.
 
+        When root dataset is specified as `id` with `recursive`, it will destroy all the children of the
```

Comment:
```suggestion
        When root dataset is specified as `id_` with `recursive`, it will destroy all the children of the
```

Reply:
We would be using `id` because that is what schema reflects. It does not matter what we are using in the actual parameter name in the function.